libraries/AP_HAL_PX4/HAL_PX4_Class.cpp

```diff
@@ -19,6 +19,12 @@
 #include "GPIO.h"
 #include "I2CDevice.h"
 #include "SPIDevice.h"
+#include "CAN.h"
+
+#include <AP_HAL_Empty/AP_HAL_Empty.h>
+#include <AP_HAL_Empty/AP_HAL_Empty_Private.h>
```

Comment:
There's no need for CAN.h here (it's not used anywhere), but if you want to leave it, you need to protect it with HAL_WITH_UAVCAN or this file won't compile for boards without CAN.